rulesets/resources/ES_ZERHruleset.rb

```diff
@@ -2528,57 +2528,32 @@ def self.get_default_ceiling_fan_cfm_per_w()
     return 122.0 # CFM per Watts
   end
 
-  def self.get_reference_ceiling_ufactor()
-    # Ceiling U-Factor
-    if [ESConstants.SFNationalVer3_0].include? @program_version
-      if ['1A', '1B', '1C', '2A', '2B', '2C', '3A', '3B', '3C'].include? @iecc_zone
-        return 0.035
-      elsif ['4A', '4B', '4C', '5A', '5B', '5C'].include? @iecc_zone
-        return 0.030
-      elsif ['6A', '6B', '6C', '7', '8'].include? @iecc_zone
-        return 0.026
-      end
-    elsif [ESConstants.SFNationalVer3_1].include? @program_version
-      if ['1A', '1B', '1C'].include? @iecc_zone
-        return 0.035
-      elsif ['2A', '2B', '2C', '3A', '3B', '3C'].include? @iecc_zone
-        return 0.030
-      elsif ['4A', '4B', '4C', '5A', '5B', '5C', '6A', '6B', '6C', '7', '8'].include? @iecc_zone
-        return 0.026
-      end
-    elsif [ESConstants.SFNationalVer3_2, ESConstants.MFNationalVer1_2].include? @program_version
-      if ['1A', '1B', '1C'].include? @iecc_zone
-        return 0.035
-      elsif ['2A', '2B', '2C', '3A', '3B', '3C'].include? @iecc_zone
-        return 0.026
-      elsif ['4A', '4B', '4C', '5A', '5B', '5C', '6A', '6B', '6C', '7', '8'].include? @iecc_zone
-        return 0.024
-      end
-    elsif [ESConstants.SFPacificVer3_0].include? @program_version
-      return 0.035
-    elsif [ESConstants.SFFloridaVer3_1].include? @program_version
-      return 0.035
-    elsif [ESConstants.SFOregonWashingtonVer3_2, ESConstants.MFOregonWashingtonVer1_2].include? @program_version
-      return 0.026
-    elsif [ESConstants.MFNationalVer1_0].include? @program_version
-      return 0.027
-    elsif [ESConstants.MFNationalVer1_1].include? @program_version
-      if ['1A', '1B', '1C', '2A', '2B', '2C', '3A', '3B', '3C', '4A', '4B'].include? @iecc_zone
-        return 0.027
-      elsif ['4C', '5A', '5B', '5C', '6A', '6B', '6C', '7', '8'].include? @iecc_zone
-        return 0.021
-      end
-    elsif [ZERHConstants.Ver1].include? @program_version
-      if ['1A', '1B', '1C'].include? @iecc_zone
-        return 0.035
-      elsif ['2A', '2B', '2C', '3A', '3B', '3C'].include? @iecc_zone
-        return 0.030
-      elsif ['4A', '4B', '4C', '5A', '5B', '5C', '6A', '6B', '6C', '7', '8'].include? @iecc_zone
-        return 0.026
-      end
+  def self.get_reference_value(value_type, window_class=nil, window_type=nil)
+    lookup_path = File.join(File.dirname(__FILE__), '../data/es_zerh_lookup.csv')
+
+    CSV.foreach(lookup_path, headers: true) do |row|
```

Comment:
Let's read the CSV data once in main.rb and then pass it into the ruleset to prevent any impact on runtime performance.